src/nodedef.cpp

```diff
@@ -676,12 +676,12 @@ void ContentFeatures::updateTextures(ITextureSource *tsrc, IShaderSource *shdsrc
 
 	bool is_liquid = false;
 
-	u8 material_type = (alpha == 255) ?
-		TILE_MATERIAL_BASIC : TILE_MATERIAL_ALPHA;
```

Comment:
I have looked at it again and as far as I can tell, that ternary isn't really doing a great lot. All the other draw-types, except opaque liquids, do support transparency to some degree and liquids are handled separately at the end of the switch-case.